tests/Feature/EditPageTest.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+use Tests\TestCase;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+
+class EditPageTest extends TestCase
+{
+    use DatabaseMigrations;
+
+    private $centre;
+    private $user;
+    private $registration;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->centre = factory(App\Centre::class)->create();
+
+        // Create a User
+        $this->user =  factory(App\User::class)->create([
+            "name"  => "test user",
+            "email" => "testuser@example.com",
+            "password" => bcrypt('test_user_pass'),
+            "centre_id" => $this->centre->id,
+        ]);
+
+        // make centre some registrations
+        $this->registration = factory(App\Registration::class)->create([
+            "centre_id" => $this->centre->id,
+        ]);
+    }
+
+    /** @test */
+    public function itShowsAPrimaryCarerInput()
+    {
+        $pri_carer = $this->registration->family->carers->first();
+        $this->actingAs($this->user)
+            ->visit(URL::route('service.registration.edit', [ 'id' => $this->registration ]))
+            ->seeElement('input[name="carer"][value="'. $pri_carer->name .'"]')
+        ;
+    }
+
+    /** @test */
+    public function itShowsASecondaryCarerInput()
+    {
+        $this->actingAs($this->user)
+            ->visit(URL::route('service.registration.edit', [ 'id' => $this->registration ]))
+            ->seeElement('input[name="carer_adder_input"]')
+            ->seeElement('button[id="add-dob"]')
+        ;
+    }
+
+    /** @test */
+    public function itShowsAListOfSecondaryCarers()
+    {
+        // Clear the carers
+        $this->registration->family->carers()->delete();
```

Comment:
Why clear the carers?

Reply:
The default carer factory produces random 1-3 carers, so we need to reset it.
We explicitly want a known number of them for this test so that
a) i can shift off the primary carer
b) i can test that it actually has 3 left.